DPG/Tasks/AOTTrack/tagAndProbeDmesons.cxx

```diff
@@ -445,65 +618,97 @@ struct TagTwoProngDisplacedVertices {
           continue;
         }
 
-        auto pVec = RecoDecay::pVec(pVecTrackPos, pVecTrackNeg);
         auto primVtx = getPrimaryVertex(collision);
         const auto& secVtx = vertexer.getPCACandidate();
         const auto& covMatrixPCA = vertexer.calcPCACovMatrixFlat();
         std::array<float, 2> trackDcaXy{trackPos.dcaXY(), trackNeg.dcaXY()};
-        if (!isSelectedTopology(primVtx, secVtx, covMatrixPCA, pVec, trackDcaXy, channel)) {
+        if (fillTagTable) {
+          getTagInfo<doMc>(primVtx, secVtx, covMatrixPCA, pVec, trackDcaXy, channel, trackPos, trackNeg, invMass2, pdgDecayMothers, particlesMc);
           continue;
```

Comment:
do you really need it? Aren't you at the end of the iteration?

Reply:
You are right. 
I've implemented the correction.

Reply:
You are right. 
I've implemented the correction.